extensions/typescript-language-features/src/configuration/configuration.electron.ts

```diff
@@ -35,4 +38,65 @@ export class ElectronServiceConfigurationProvider extends BaseServiceConfigurati
 		}
 		return null;
 	}
+
+	protected readLocalNodePath(configuration: vscode.WorkspaceConfiguration): string | null {
+		return this.validatePath(this.readLocalNodePathWorker(configuration));
+	}
+
+	private readLocalNodePathWorker(configuration: vscode.WorkspaceConfiguration): string | null {
+		const inspect = configuration.inspect('typescript.tsserver.nodePath');
+		if (inspect?.workspaceValue && typeof inspect.workspaceValue === 'string') {
+			if (inspect.workspaceValue === 'node') {
+				return this.findNodePath();
+			}
+			const fixedPath = this.fixPathPrefixes(inspect.workspaceValue);
+			if (!path.isAbsolute(fixedPath)) {
+				const workspacePath = RelativeWorkspacePathResolver.asAbsoluteWorkspacePath(fixedPath);
+				return workspacePath || null;
+			}
+			return fixedPath;
+		}
+		return null;
+	}
+
+	protected readGlobalNodePath(configuration: vscode.WorkspaceConfiguration): string | null {
+		return this.validatePath(this.readGlobalNodePathWorker(configuration));
+	}
+
+	private readGlobalNodePathWorker(configuration: vscode.WorkspaceConfiguration): string | null {
+		const inspect = configuration.inspect('typescript.tsserver.nodePath');
+		if (inspect?.globalValue && typeof inspect.globalValue === 'string') {
+			if (inspect.globalValue === 'node') {
+				return this.findNodePath();
+			}
+			const fixedPath = this.fixPathPrefixes(inspect.globalValue);
+			if (path.isAbsolute(fixedPath)) {
+				return fixedPath;
+			}
+		}
+		return null;
+	}
+
+	private findNodePath(): string | null {
+		try {
+			const out = child_process.execFileSync('node', ['-e', 'console.log(process.execPath)'], {
```

Comment:
Context: I tried doing the node detection using `which`, like other language extensions do, but it does not work if you have a node version manager like volta, because if you do, your path points to volta's `node.exe` wrapper, and if you use that to run TS Server, it crashes.
This alternative approach was suggested by @jakebailey.

Reply:
What do you mean for all paths?